db/version_set.h

```diff
@@ -1501,6 +1507,11 @@ class VersionSet {
         new_options.writable_file_max_buffer_size;
   }
 
+  const OffpeakTimeInfo& offpeak_time_info() { return offpeak_time_info_; }
+  void ChangeOffpeakTimeInfo(const std::string& daily_offpeak_time_utc) {
```

Comment:
Since this should be kept consistent with the CF option, like writable_file_max_buffer_size, I'd rather see just one function in VersionSet for updating its fields to match MutableDBOptions. And it should be clear what the synchronization strategy/requirement is (requires DB mutex held?).

I guess you might still want a simple setter for unit tests, but that could be called TEST_whatever to discourage haphazard use. Or you could set it indirectly with MutableCFOptions.

Reply:
Agreed, but I was a bit surprised to see that we update options in `SetDBOptions()` in a way that updates may not be consistent across all options. For instance, `max_background_flushes`, `max_background_compactions`, `max_background_jobs` get updated first and compactions get scheduled again before updating other fields like `max_total_wal_size`, `compaction_readahead_size`). https://github.com/facebook/rocksdb/blob/526f36b48381dd640a0426bd748dbc0bb5797c75/db/db_impl/db_impl.cc#L1343

So I figured `daily_offpeak_time_utc ` may go with options like `max_background_compactions` and updated along with them.

Reply:
TODO seems reasonable for now, thanks